hclsyntax/expression_test.go

```diff
@@ -2174,8 +2224,12 @@ EOT
 	for _, test := range tests {
 		t.Run(test.input, func(t *testing.T) {
 			expr, parseDiags := ParseExpression([]byte(test.input), "", hcl.Pos{Line: 1, Column: 1, Byte: 0})
+			var got cty.Value
+			var valDiags hcl.Diagnostics
 
-			got, valDiags := expr.Value(test.ctx)
+			if expr != nil {
```

Comment:
I'm curious to learn if `ParseExpression` returned `nil` before this change. We do not currently check this in terraform-ls and would need to handle this case.

Reply:
The general convention in HCL is that if a function returns error diagnostics then other return values might be somehow incomplete. 

Being totally `nil` is an extreme case of being incomplete, and I would suggest callers consider it to mean "this was so invalid that it doesn't make sense to even partially describe it".